api/v1beta1/openstackbaremetalset.go

```diff
@@ -129,16 +137,23 @@ func VerifyBaremetalSetScaleUp(
 
 		// If we can't satisfy the new requested BaremetalHost count, explicitly state so
 		if newBmhsNeededCount > len(availableBaremetalHosts) {
-			return nil, fmt.Errorf("Unable to find %d requested BaremetalHosts for scale-up (%d in use, %d available)",
+			errLabelStr := ""
+
+			if labelStr != "" {
+				errLabelStr = fmt.Sprintf(" with labels %s", labelStr)
+			}
+
+			return nil, fmt.Errorf("Unable to find %d requested BaremetalHosts%s in namespace %s for scale-up (%d in use, %d available)",
```

Comment:
with label %s

Reply:
Maybe `with label(s) %s` since the label selector can have more than one label in it, but might only have one.